.github/workflows/test.yml

```diff
@@ -1,52 +1,38 @@
-## test.yml
+name: CI
 
-name: Main workflow
-on: [push, pull_request]
+on:
+  pull_request:
+    paths-ignore:
+    - '**.md'
+    - '**.rst'
+  push:
+    paths-ignore:
+    - '**.md'
+    - '**.rst'
+    branches-ignore:
+    - 'master'
+    - 'main'
 
 jobs:
   build:
     runs-on: ubuntu-latest
     strategy:
-      fail-fast: false
```

Comment:
Please keep this fail-fast config? Please tell me a reason to remove this?

Reply:
Restored 37052df